kernel/userspace.c

```diff
@@ -409,6 +409,66 @@ void k_object_free(void *obj)
 	}
 }
 
+int z_impl_k_alloc_thread_stack(size_t size, k_thread_stack_t **stack)
+{
+	struct z_object *zo;
+	struct dyn_obj *dyn;
+	struct dyn_obj *adjusted_dyn;
+	k_spinlock_key_t key;
+	const size_t align = Z_THREAD_STACK_OBJ_ALIGN(size);
+	/* We want to append the dyn_obj to the end of the stack area. However,
```

Comment:
That sounds wrong to me.  The struct z_object is a kernel struct.  It has privileged fields and needs to be in kernel space, inaccessible to userspace threads.  The stack you're returning is (by definition) going to be written by a user thread.  These surely want to be separate blocks.